packages/web/src/components/EmailConfirmMessage/EmailConfirmMessage.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { AlertPlaceholder, ErrorAlert, InfoAlert } from 'elements/alerts';
+import { QueryParams } from '../../constants/query-params';
+
+type Props = {
+  isLoading: boolean;
+  error: string | null;
+  onInit: (token: string) => void;
+}
+
+export const EmailConfirmMessage = ({ onInit, error, isLoading }: Props) => {
+  const location = useLocation();
+
+  useEffect(() => {
+    const params = new URLSearchParams(location.search);
+    onInit(params.get(QueryParams.EMAIL_VERIFICATION_TOKEN) as string);
+  }, []);
```

Comment:
The warning you see from `exhaustive-deps` is saying that you're trying to apply an old mental model to (`componentDidMount()`) to hooks. And is also trying to save you from some bugs when updates are missed because you forget to include the dependency. In most of the case you shouldn't be afraid to include the dependency if you don't expect it to change. Like in this case, is there a chance location.search will change while the component lives?